gctscript/modules/gct/common_test.go

```diff
@@ -0,0 +1,174 @@
+			&objects.Float{Value: 4},
+			&objects.Float{Value: 5},
+		},
+	}
+
+	threeElement = objects.Array{
+		Value: []objects.Object{
+			&objects.Array{
+				Value: []objects.Object{
+					&objects.Float{Value: 11},
+					&objects.Float{Value: 12},
+					&objects.Float{Value: 13},
+				},
+			},
+			&objects.Array{
+				Value: []objects.Object{
+					&objects.Float{Value: 21},
+					&objects.Float{Value: 22},
+					&objects.Float{Value: 23},
+				},
+			},
+			&objects.Array{
+				Value: []objects.Object{
+					&objects.Float{Value: 31},
+					&objects.Float{Value: 32},
+					&objects.Float{Value: 33},
+				},
+			},
+			&objects.Array{
+				Value: []objects.Object{
+					&objects.Float{Value: 41},
+					&objects.Float{Value: 42},
+					&objects.Float{Value: 43},
+				},
+			},
+			&objects.Array{
+				Value: []objects.Object{
+					&objects.Float{Value: 51},
+					&objects.Float{Value: 52},
+					&objects.Float{Value: 53},
+				},
+			},
+		},
+	}
+
+	ohlcv = []objects.Object{
+		&objects.Time{Value: time.Now()},
+		&objects.Float{Value: 100},
+		&objects.Float{Value: 100},
+		&objects.Float{Value: 100},
+		&objects.Float{Value: 100},
+		&objects.Float{Value: 1},
+	}
+
+	ohlcdata = objects.Map{
+		Value: map[string]objects.Object{
+			"exchange":  &objects.String{Value: "exchange"},
+			"pair":      &objects.String{Value: "BTC-USD"},
+			"asset":     &objects.String{Value: asset.Spot.String()},
+			"intervals": &objects.String{Value: time.Minute.String()},
+			"candles": &objects.Array{
+				Value: []objects.Object{
+					&objects.Array{
+						Value: ohlcv,
+					},
+					&objects.Array{
+						Value: ohlcv,
+					},
+					&objects.Array{
+						Value: ohlcv,
+					},
+					&objects.Array{
+						Value: ohlcv,
+					},
+					&objects.Array{
+						Value: ohlcv,
+					},
+				},
+			},
+		},
+	}
+)
+
+func TestCommonWriteToCSV(t *testing.T) {
+	t.Parallel()
+
+	OutputDir = filepath.Join(os.TempDir(), "script-temp")
+
+	_, err := WriteAsCSV()
+	if err == nil {
+		t.Fatal("error cannot be nil")
+	}
+
+	_, err = WriteAsCSV(nil)
+	if err == nil {
+		t.Fatal("error cannot be nil")
+	}
+
+	_, err = WriteAsCSV(&objects.String{Value: "something.txt"})
+	if err == nil {
+		t.Fatal(err)
+	}
+
+	_, err = WriteAsCSV(&objects.String{Value: "something.txt"},
+		&objects.String{Value: "extra string"})
+	if err == nil {
+		t.Fatal(err)
+	}
+
+	_, err = WriteAsCSV(&objects.String{Value: "script-temp.csv"}, unhandled)
+	if err == nil {
+		t.Fatal("error cannot be nil")
+	}
+
+	_, err = WriteAsCSV(&objects.String{Value: "script-temp.csv"},
+		atrPayload,
+		bbandsPayload,
+		emaPayload,
+		macdPayload,
+		mfiPayload,
+		obvPayload,
+		rsiPayload,
+		smaPayload,
+		ohlcPayload)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, err = WriteAsCSV(atrPayload)
+	if err == nil {
+		t.Fatal(err)
+	}
+
+	_, err = WriteAsCSV(&objects.String{Value: "test.gct-script-temp2"},
+		atrPayload)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = os.RemoveAll(OutputDir)
```

Comment:
Can add this as a defer cleaup() func as the other writetocsv funcs could potentially fail without cleaning up after itself.

Reply:
🦅